assets/js/components/settings/SettingsModules.js

```diff
@@ -19,283 +19,84 @@
-						mdc-layout-grid__cell--span-12
-					">
-						<Layout>
-							{ activeModules }
-						</Layout>
-					</div>
-				}
-
-				{ /* Inactive Modules */ }
-				{ 1 === activeTab && // If <SettingsApp/> is on the Add tab. TODO this could be removed after refactoring this into separate components.
-					inactiveModulesAvailable && // We have inactive modules available.
-
-					<div className="
-								mdc-layout-grid__cell
-								mdc-layout-grid__cell--span-12
-							">
-						<Layout
-							header
-							title={ __( 'Connect More Services to Gain More Insights', 'google-site-kit' ) }
-							relative
-						>
-							<div className="mdc-layout-grid">
-								<div className="mdc-layout-grid__inner">
-									{ 0 < inactiveModules.length && inactiveModules }
-								</div>
-							</div>
-							{ /* TODO: Need some input here with regards to changing state */ }
-							{ editActive && <SettingsOverlay /> }
-						</Layout>
-					</div>
-				}
-				{ 1 === activeTab && // If <SettingsApp/> is on the Add tab. TODO this could be removed after refactoring this into separate components.
-					! inactiveModulesAvailable && // If we have no active modules.
-					<div className="
-						mdc-layout-grid__cell
-						mdc-layout-grid__cell--span-12
-					">
-						<Notification
-							id="no-more-modules"
-							title={ __( 'Congrats, you’ve connected all services!', 'google-site-kit' ) }
-							description={ __( 'We’re working on adding new services to Site Kit by Google all the time, so please check back in the future.', 'google-site-kit' ) }
-							format="small"
-							smallImage={ global._googlesitekitLegacyData.admin.assetsRoot + thumbsUpImage }
-							type="win-success"
-						/>
-					</div>
-				}
-			</Fragment>
-		);
-	}
+	const toDependantModules = ( dependantModules, dependantSlug ) => {
+		const dependantModule = modulesData[ dependantSlug ];
+		return dependantModule
+			?	[ ...dependantModules, dependantModule.name ]
+			: dependantModules;
+	};
+
+	const getDependantModulesText = ( { dependants } ) =>
+		dependants && dependants.length > 0
+			? listFormat( dependants.reduce( toDependantModules, [] ) )
+			: '';
+
+	const withDependantModulesText = ( module ) => ( {
+		...module,
+		dependantModulesText: getDependantModulesText( module ),
+	} );
+
+	const byActiveNoInternals = ( active ) => ( module ) =>
+		! module.internal && active === module.active;
+
+	const getModulesByActive = ( active ) =>
+		Object.values( modulesData )
+			.filter( byActiveNoInternals( active ) )
+			.sort( ( a, b ) => a.sort - b.sort )
+			.map( withDependantModulesText );
+
+	const activeModules = getModulesByActive( true );
+	const inactiveModules = getModulesByActive( false );
+
+	return (
+		<Switch>
+			{ /* Settings Module Routes */ }
+			<Route path={ [ '/connected-services/:moduleSlug/:action', '/connected-services/:moduleSlug', '/connected-services' ] }>
+				<SettingsConnectedServices modules={ activeModules } />
+			</Route>
```

Comment:
Hmm, I wonder what others thoughts are on this? I'd usually make multiple routes with different `path` values but the same `children` for things like this, eg:

```suggestion
			{ /* Settings Module Routes */ }
			<Route path="/connected-services/:moduleSlug/:action">
				<SettingsConnectedServices modules={ activeModules } />
			</Route>
			<Route path="/connected-services/:moduleSlug">
				<SettingsConnectedServices modules={ activeModules } />
			</Route>
			<Route path="/connected-services">
				<SettingsConnectedServices modules={ activeModules } />
			</Route>
```

I suppose the downside is that it might re-render the entire child/mount a new component with similar props when we change actions, etc.? Is that what this is trying to avoid?

No strong opinion here, but I do like the readability of a path-per-Route approach.

cc @aaemnnosttv @felixarntz

Reply:
I tend to like the flatter path-per-route way as well, although it does look a bit odd that the result is the same for each. Would it be non-idiomatic to pass the props through directly here using one of the render prop variants of `Route`?

Reply:
@aaemnnosttv @tofumatt Modified this section to use flat paths. It will re-mount on change, but it doesn't appear to be an issue. I attempted to use render props but it still requires an array value for the path prop or multiple routes (flat paths).